### PR TITLE
dae,nixos/dae: 0.2.4 -> 0.3.0

### PR DESCRIPTION
**Pull request by oluceps**
## Description of changes
diff: https://github.com/daeuniverse/dae/compare/v0.2.4...v0.3.0

Sync nixos module from https://github.com/daeuniverse/flake.nix .

Picked a commit from pr https://github.com/NixOS/nixpkgs/pull/251161 since the option exists on upstream.
Mentioning @zzzsyyy

Removed the `wrapProgram` to avoid potential performance cost. Dependencies added in the module by default, and a option for user to easy specify assets. 

https://github.com/daeuniverse/flake.nix/pull/11

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by pokon548 — APPROVED** [review]
Looks great for me 💯

File: nixos/modules/services/networking/dae.nix [hunk 1]
```diff
@@ -1,41 +1,159 @@
...
+          Open the firewall port.
+        '';
+      };
+
+      configFile = mkOption {
+        type = types.path;
+        default = "/etc/dae/config.dae";
+        example = "/path/to/your/config.dae";
+        description = mdDoc ''
+          The path of dae config file, end with `.dae`.
```
Comment (zzzsyyy) [inline nixos/modules/services/networking/dae.nix:77]: How about adding a link for a example conf here?

```suggestion
          The path of dae config file, end with `.dae`.
          
          See <https://github.com/daeuniverse/dae/blob/main/example.dae>.
```

File: nixos/modules/services/networking/dae.nix [hunk 2]
```diff
@@ -1,41 +1,159 @@
...
+      };
+
+      config = mkOption {
+        type = types.str;
+        default = ''
+          global{}
+          routing{}
+        '';
+        description = mdDoc ''
+          Config text for dae.
```
Comment (zzzsyyy) [inline nixos/modules/services/networking/dae.nix:88]:
```suggestion
          Config text for dae.
          
          See <https://github.com/daeuniverse/dae/blob/main/example.dae>.
```

**Review by zzzsyyy — APPROVED** [review]

**Review by pokon548 — APPROVED** [review]

**Review by pokon548 — APPROVED** [review]

**Comment by luochen1990** [comment]
LGTM.

**Pull request merged**